src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -880,7 +909,34 @@ void QgsRelationReferenceWidget::filterChanged()
 
 void QgsRelationReferenceWidget::addEntry()
 {
-  QgsFeature f( mReferencedLayer->fields() );
+  if ( !mReferencedLayer )
+    return;
+
+  const QgsVectorLayerTools *tools = mEditorContext.vectorLayerTools();
+  if ( !tools )
+    return;
+  if ( !mCanvas )
+    return;
+
+  mMapToolDigitize->setLayer( mReferencedLayer );
+  setMapTool( mMapToolDigitize );
+
+  connect( mMapToolDigitize, static_cast<void ( QgsMapToolDigitizeFeature::* )( const QgsFeature & )>( &QgsMapToolDigitizeFeature::digitizingCompleted ), this, &QgsRelationReferenceWidget::entryAdded );
```

Comment:
can you check if `qgis::overload<T>` works for you here?

Reply:
Actually, there is no need of `qgis::overload` or `static_cast<>` because there is only one version of the signal. I correct it. I wasn't aware of this problem, I see this way of connecting signal in the file and thought it was a coding guideline. I learn something today :) thanks!

Reply:
If you find any static_cast on connections, feel free to replace them with `qgis::overload<T>` it's much more readable and it will be easily replaced with `qOverload` when we will raise QT min version.

Reply:
OK :ok_hand: